step/v2/examples/simple/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/pulumi/upgrade-provider/step/v2"
+)
+
+func main() {
+	ctx := context.Background()
+	if file := os.Getenv("STEP_RECORD"); file != "" {
+		var closer io.Closer
+		ctx, closer = step.WithRecord(ctx, file)
+		defer func() {
+			if err := closer.Close(); err != nil {
+				panic(err)
+			}
+		}()
+	}
+
+	if err := step.PipelineCtx(ctx, "simple", pipeline); err != nil {
+		os.Exit(1)
+	}
+}
+
+func pipeline(ctx context.Context) {
+	bytes := step.ReadFile(ctx, "input.txt")
```

Comment:
Suppose you wanted to handle an error (eg. try to open this file, if it fails open this other file) how would you do it?

Reply:
You can't do that with `step.ReadFile`, but `step.ReadFile` isn't special. It's implementation looks like this:

https://github.com/pulumi/upgrade-provider/blob/2d789477512e2dc4b5c2c3d1161340d43d01c31b/step/v2/util.go#L18-L26

Because go `error`s are just values, `ReadFileOrError` would look like this:

```go
// Read a file, halting the pipeline on error.
func ReadFileOrError(ctx context.Context, path string) (string, error) {
	return Call12(ctx, path, func(ctx context.Context, path string) (string, error) {
		MarkImpure(ctx)
		bytes, err := os.ReadFile(path)
		SetLabel(ctx, fmt.Sprintf("%d bytes read", len(bytes)))
		return string(bytes), err
	}, path)
}
```

Reply:
Ah, ok, thanks for clarifying; possibly worth exposing both versions of steps in our "stdlib", but seeing this resolves my concern about a structural limitation

Reply:
Ah, ok, thanks for clarifying; possibly worth exposing both versions of steps in our "stdlib", but seeing this resolves my concern about a structural limitation